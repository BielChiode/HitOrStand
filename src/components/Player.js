import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'
import { Icon } from 'react-native-elements'
import sumPoints from '../utils/sumPoints'
import PropTypes from 'prop-types'

function Player({ player }) {
  const points = sumPoints(player)

  return (
    <View style={styles.playerContainer}>
      <View style={styles.dealerAvatar}>
        <Icon raised name="user" type="font-awesome" color="#99CFBD" />
        <Text style={styles.textPoints}>Pontos: {points}</Text>
      </View>
      <View style={styles.playerCardsContainer}>
        {player &&
          player.map((card, index) => (
            <Image
              key={index}
              style={[styles.cardImage, { marginLeft: index !== 0 ? -50 : 0 }]}
              source={{
                uri: card.image
              }}
            />
          ))}
      </View>
    </View>
  )
}

Player.propTypes = {
  player: PropTypes.array.isRequired
}

export default Player

const styles = StyleSheet.create({
  playerContainer: {
    position: 'absolute',
    width: '100%',
    bottom: 0
  },
  playerCardsContainer: {
    flexDirection: 'row',
    marginBottom: 16
  },
  dealerAvatar: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  textPoints: {
    fontSize: 20,
    fontWeight: '600',
    marginLeft: 10,
    color: '#FFF'
  },
  cardImage: {
    width: 110,
    height: 150
  }
})
