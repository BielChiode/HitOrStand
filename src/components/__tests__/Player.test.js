import { render } from '@testing-library/react-native'
import Player from '../Player'

describe('Player', () => {
  const player = [
    {
      code: '5D',
      image: 'https://deckofcardsapi.com/static/img/5D.png',
      suit: 'DIAMONDS',
      value: '5'
    },
    {
      code: '4C',
      image: 'https://deckofcardsapi.com/static/img/4C.png',
      suit: 'CLUBS',
      value: '4'
    }
  ]
  it('Deveria renderizar corretamente com as cartas e com pontos', () => {
    const { toJSON, getByText } = render(<Player player={player} />)
    expect(getByText('Pontos: 9'))
    expect(toJSON()).toMatchSnapshot()
  })
})
