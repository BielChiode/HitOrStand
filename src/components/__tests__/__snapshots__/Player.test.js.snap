// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Deveria renderizar corretamente com as cartas e com pontos 1`] = `
<View
  style={
    {
      "bottom": 0,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 28,
          "height": 52,
          "margin": 7,
          "overflow": "hidden",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 52,
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "height": 52,
              "justifyContent": "center",
              "width": 52,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#99CFBD",
                  "fontSize": 24,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            ï€‡
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
    >
      Pontos: 
      9
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 16,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/5D.png",
        }
      }
      style={
        [
          {
            "height": 150,
            "width": 110,
          },
          {
            "marginLeft": 0,
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/4C.png",
        }
      }
      style={
        [
          {
            "height": 150,
            "width": 110,
          },
          {
            "marginLeft": -50,
          },
        ]
      }
    />
  </View>
</View>
`;
